clang/test/Driver/sycl-offload.c

```diff
@@ -182,6 +182,11 @@
 // CHK-PHASES-LIB: 15: clang-offload-wrapper, {14}, object, (device-sycl)
 // CHK-PHASES-LIB: 16: offload, "host-sycl (x86_64-unknown-linux-gnu)" {10}, "device-sycl (spir64-unknown-linux-sycldevice)" {15}, image
 
+/// Compilation check with -lstdc++ (treated differently than regular lib)
+// RUN:   %clang -### -target x86_64-unknown-linux-gnu -lstdc++ -fsycl -fsycl-targets=spir64-unknown-linux-sycldevice %s 2>&1 \
+// RUN:   | FileCheck -check-prefix=CHK-LIB-STDCXX %s
+// CHK-LIB-STDCXX: ld{{.*}} "-lstdc++"
```

Comment:
I'd suggest adding some `CHK-LIB-STDCXX-NOT`s for device actions

Reply:
Added some device checks